File: 백준/Silver/1966. 프린터 큐/프린터 큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

    private static int t;
    private static int n;
    private static int m;
    private static Queue<Integer> importance; //문서의 중요도
    private static Queue<Integer> index; //문서의 인덱스

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        t = Integer.parseInt(br.readLine());

        for(int i = 0; i < t; i++) {

            StringTokenizer st = new StringTokenizer(br.readLine());
            n = Integer.parseInt(st.nextToken());
            m = Integer.parseInt(st.nextToken());

            importance = new LinkedList<>();
            index = new LinkedList<>();

            st = new StringTokenizer(br.readLine());
            for(int j = 0; j < n; j++) {
                importance.offer(Integer.parseInt(st.nextToken()));
                index.offer(j);
            }
            solveAndPrintResult();
        }
    }
    
    private static void solveAndPrintResult() {
        int count = 1;
        
        while(!importance.isEmpty()) {
            int maxImportance = Collections.max(importance);
            int curImportance = importance.poll();
            int curIndex = index.poll();
            
            if(curImportance == maxImportance) {
                if(curIndex == m) {
                    System.out.println(count);
                    break;
                }
                count++;
            }
            else {
                importance.offer(curImportance);
                index.offer(curIndex);
            }
        }
    }
}

File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
from queue import Queue

def solve_and_print_result(n, m, importance_values):
    importance = Queue()
    index = Queue()

    for j, val in enumerate(importance_values):
        importance.put(val)
        index.put(j)

    count = 1

    while not importance.empty():
        max_importance = max(importance.queue)
        cur_importance = importance.get()
        cur_index = index.get()

        if cur_importance == max_importance:
            if cur_index == m:
                print(count)
                break
            count += 1
        else:
            importance.put(cur_importance)
            index.put(cur_index)

t = int(input())

for _ in range(t):
    n, m = map(int, input().split())
    importance_values = list(map(int, input().split()))
    
    solve_and_print_result(n, m, importance_values)
